src/jit/lower.cpp

```diff
@@ -3737,9 +3746,15 @@ void Lowering::LowerBlock(BasicBlock* block)
     m_block = block;
     m_blockRange = LIR::AsRange(block);
 
-    for (GenTree* node = m_blockRange.FirstNonPhiNode(), *end = m_blockRange.End(); node != end; node = node->gtNext)
+    for (GenTree* node = m_blockRange.FirstNonPhiNode(), *end = m_blockRange.End(), *next; node != end; node = next)
     {
-        LowerNode(node);
+        node = LowerNode(node);
+
```

Comment:
I agree that the two passes should use the same model. Personally, I prefer returning the lowered/decomposed node to returning the next node to process, but I'm not super committed to it. Do you have a preference?